hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4123,10 +4128,37 @@ func reconcileOpenshiftMonitoringNetworkPolicy(policy *networkingv1.NetworkPolic
 	return nil
 }
 
-func reconcileVirtLauncherNetworkPolicy(policy *networkingv1.NetworkPolicy, hcluster *hyperv1.HostedCluster) error {
+func addToBlockedNetworks(network string, blockedIPv4Networks []string, blockedIPv6Networks []string) ([]string, []string) {
+	if utilsnet.IsIPv6CIDRString(network) {
+		blockedIPv6Networks = append(blockedIPv6Networks, network)
+	} else {
+		blockedIPv4Networks = append(blockedIPv4Networks, network)
+	}
+	return blockedIPv4Networks, blockedIPv6Networks
+}
+
+func reconcileVirtLauncherNetworkPolicy(policy *networkingv1.NetworkPolicy, hcluster *hyperv1.HostedCluster, managementClusterNetwork *configv1.Network) error {
 	protocolTCP := corev1.ProtocolTCP
 	protocolUDP := corev1.ProtocolUDP
 	protocolSCTP := corev1.ProtocolSCTP
+
+	blockedIPv4Networks := []string{}
+	blockedIPv6Networks := []string{}
+	for _, network := range managementClusterNetwork.Spec.ClusterNetwork {
+		blockedIPv4Networks, blockedIPv6Networks = addToBlockedNetworks(network.CIDR, blockedIPv4Networks, blockedIPv6Networks)
+	}
+
+	for _, network := range managementClusterNetwork.Spec.ServiceNetwork {
+		blockedIPv4Networks, blockedIPv6Networks = addToBlockedNetworks(network, blockedIPv4Networks, blockedIPv6Networks)
+	}
```

Comment:
One thing to double check. are we sure this doesn't impact the ability to route to and from LoadBalancers created by cloud-provider-kubevirt?

For example. lets say within the guest OCP i create a LB service that points to some pod within the guest cluster... Will clients within other pods on the guest cluster be able to contact that LB, or will that traffic be unintentionally blocked due to blocking off the service ip range on the infra cluster?

From what I gather, i think this scenario will work because the virt-launcher pods are in the allow list, but i'm not completely certain.

Reply:
I confirmed that with this new NetworkPolicy in place, a pod on the guest cluster that is exposed via a LoadBalancer service can be reached from another pod on the guest cluster, also if it is running on another guest node (virt-launcher on HCP).

Also, all `[sig-kubevirt]` conformance test cases are passing with this NetworkPolicy.

The explanation is that we're allowing traffic between the virt-launcher to each other, as you said.